### PR TITLE
Feat/rpc tests

### PR DESCRIPTION
**Pull request by willemneal**
### What

Adds NetworkRunnable trait so that commands that need the RPC can have a common run function. 

Adds to the bindings action to test RPC.

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

[TODO or N/A]

### CODE REVIEW

**Comment by janewang** [comment]
Blocked. Needs on help on this failure: https://github.com/stellar/soroban-tools/actions/runs/7921583339/job/21627229312?pr=1202 and also which quickstart to point to

**Comment by janewang** [comment]
Still blocked. Tests are flaky. This blocks https://github.com/stellar/soroban-cli/issues/1106

**Comment by willemneal** [comment]
I was able to fix the rpc tests with some sleeps (which suggests a concurrency issue), but it seems that I shouldn't use async traits in one of the msrv tests because it's a recent addition to rust. Is this an issue? I could probably use the crate, but last time I tried (a year ago) I ran into some issues.
@leighmcculloch

File: cmd/crates/soroban-test/tests/it/integration/dotenv.rs [hunk 1]
```diff
@@ -1,67 +1,68 @@
...
+    std::fs::write(&env_file, &contents).unwrap();
     assert_eq!(contents, std::fs::read_to_string(env_file).unwrap());
 }
 
-fn contract_id() -> String {
-    format!("SOROBAN_CONTRACT_ID={TEST_CONTRACT_ID}")
+#[tokio::test]
+async fn can_read_file() {
+    let e = &TestEnv::new();
+    std::thread::sleep(core::time::Duration::from_millis(1000));
```
Comment (leighmcculloch) [inline cmd/crates/soroban-test/tests/it/integration/dotenv.rs:15]: Do you have a sense of what's causing the need for this delay? Is it waiting for the quickstart image to start up before deploying, or something else?

**Review by leighmcculloch — APPROVED** [review]
It's not ideal to keep those sleeps in, but also don't want to block us moving forward with this and getting this committed. I do think we need to get to the bottom of why the sleeps are needed though.

File: cmd/crates/soroban-test/tests/it/integration/dotenv.rs (hunk 1 above)
Comment (willemneal) [reply cmd/crates/soroban-test/tests/it/integration/dotenv.rs:15]: Nope the health check ensures it's ready and the test has funded an account by this point. So still a bit stumped. Will get more logging info tomorrow.

File: cmd/crates/soroban-test/src/lib.rs [hunk 2]
```diff
@@ -53,11 +57,16 @@ pub enum Error {
...
+    pub rpc_url: String,
 }
 
 impl Default for TestEnv {
     fn default() -> Self {
-        Self::new().unwrap()
+        let temp_dir = TempDir::new().unwrap();
+        Self {
+            temp_dir,
+            rpc_url: "http://localhost:8889/soroban/rpc".to_string(),
```
[the remark this answers is not in the document]
Comment (willemneal) [reply cmd/crates/soroban-test/src/lib.rs:68]: SOROBAN_PORT is used.

File: cmd/crates/soroban-test/tests/it/integration/dotenv.rs (hunk 1 above)
Comment (leighmcculloch) [reply cmd/crates/soroban-test/tests/it/integration/dotenv.rs:15]: Got it. Feel free to merge and follow up.

Comment (willemneal) [reply cmd/crates/soroban-test/tests/it/integration/dotenv.rs:15]: https://github.com/stellar/soroban-cli/issues/1231

**Pull request merged**